Formula/mysql-client.rb

```diff
@@ -35,6 +35,10 @@ class MysqlClient < Formula
 
   fails_with gcc: "5"
 
+  # Fix for "Cannot find system zlib libraries" even though they are installed.
+  # https://bugs.mysql.com/bug.php?id=110745
+  patch :DATA
```

Comment:
We should move this to Homebrew/formula-patches if we end up carrying this for another release.

Reply:
Upstream responded to that bug report today, so we should be ok:
>Fixed as of the upcoming MySQL Server 8.0.34 and 8.1.0 releases, and here's the proposed changelog entry from the documentation team:
>
>MySQL would not build with -DWITH_ZLIB=system; it'd complain about not
>finding the system zlib library despite finding it.
>
>Thank you for the bug report.